Kitodo/src/main/java/org/kitodo/production/services/index/IndexingService.java

```diff
@@ -124,23 +127,54 @@ public CompletionStage<?> startIndexing(Class<? extends BaseBean> type, MassInde
 
     /**
      * Searches for a search term in a search field and returns the hit IDs.
-     * 
+     *
      * @param beanClass
      *            class of beans to search for
      * @param searchField
      *            search field to search on
      * @param value
      *            value to be found in the search field
+     * @param useScroll
+     *            whether to use scrolling (for large result sets like exports)
      * @return ids of the found beans
      */
-    public Collection<Integer> searchIds(Class<? extends BaseBean> beanClass, String searchField, String value) {
+    public Collection<Integer> searchIds(Class<? extends BaseBean> beanClass,
+                                         String searchField,
+                                         String value,
+                                         boolean useScroll) {
+
         SearchSession searchSession = Search.session(HibernateUtil.getSession());
```

Comment:
While reading the code changes here: `HibernateUtil.getSession()` is returing an Hibernate-Session which implements `AutoCloseable` interface. Should here and the other places in this class this usage replaced by a try-with-ressources statement? I noticed that using the search beginning with 3.9.x consumes more resources than before and this could be a reason. Maybe this should be fixed in a separate pull request. I did not start a discussion why in a `IndexingService` labeled class search methods are defined and used instead doing this in a `SearchService` class like in 3.8.x.